scripts/taxonomy/metadata_models.py

```diff
@@ -0,0 +1,318 @@
+import collections
+import logging
+import json
+import os
+from isb_web import config
+
+from isamples_metadata.Transformer import Transformer
+from scripts.taxonomy.Model import Model
+from scripts.taxonomy.SESARClassifierInput import SESARClassifierInput
+from scripts.taxonomy.OpenContextClassifierInput import OpenContextClassifierInput
+
+
+class MetadataModelLoader:
+    """Class that instantiates the pretrained models"""
+
+    # loaded models that will be used for classification
+    _SESAR_MATERIAL_MODEL = None
+    _OPENCONTEXT_MATERIAL_MODEL = None
+    _OPENCONTEXT_SAMPLE_MODEL = None
+
+    @staticmethod
+    def load_model_from_path(collection, label_type):
+        """
+            Set the pretrained models by loading them from the file system
+            Prerequisite: In order to use this, make sure that there is a pydantic settings file on the
+            at the root of this repository named "isamples_web_config.env" with the config file path, and the model path
+            defined
+
+            :param collection : the collection type of the sample
+            :param label_type : the field that we want to predict
+        """
+        # load the model config file
+        if collection == "SESAR" and label_type == "material":
+            config_path = config.Settings().sesar_material_config_path
+        elif collection == "OPENCONTEXT" and label_type == "material":
+            config_path = config.Settings().opencontext_material_config_path
+        elif collection == "OPENCONTEXT" and label_type == "sample":
+            config_path = config.Settings().opencontext_sample_config_path
+        # read the model config file
+        if not os.path.exists(config_path):
+            logging.error(
+                "Unable to locate pretrained models",
+            )
+            return
+        with open(config_path) as json_file:
+            config_json = json.load(json_file)
+        # use the model config to get the pretrained model
+        model = Model(config_json)
+
+        # initialize the model fields
```

Comment:
I would expect this to not re-instantiate the model and open the JSON every time it is called.  Is there somewhere else that prevents that from happening?

Reply:
Ah, I see, it looks like `initialize_models` is responsible for that.